### DISCUSSION TITLE
OTAA join procedure

### DISCUSSION THREAD

**Discussion by TheStudent227** [Lora-net/LoRaMac-node#1176] [Q&A]
Hello, I would like to ask for help with LoRa network server. I am doing the project on CoDeSys for controller, which should emulate the network server. The end device is premade, but I have troubles finding examples of network server code which will handle the request from LoRa sensor. Is there some available network server code ?

Apologizing for possible off-topic

**Comment by mluis1** [comment]
This project is only related to the end-device side.

For the network server side there are multiple open source projects. The ones that I am aware of are:
- https://www.chirpstack.io/ [GO language]
- https://github.com/TheThingsNetwork/lorawan-stack [GO language]
- https://github.com/gotthardp/lorawan-server [Erlang language]
